source/core/NumericString.cpp

```diff
@@ -1953,6 +1954,97 @@ VIREO_FUNCTION_SIGNATUREV(StringScan, StringScanParamBlock)
+    return true;
+}
+
+static bool IsValidUTF8AndUpdateErrorAndMakeOutputArrayEmpty(UInt8* begin, UInt8* end, ErrorCluster* errorCluster, TypedArrayCoreRef outputArray)
+{
+    if (!utf8::is_valid(begin, end)) {
+        if (errorCluster != nullptr) {
+            errorCluster->SetError(true, 1, "Invalid UTF-8 characters found");  // TODO(sanmut): Update error code.
+        }
+        if (outputArray != nullptr)
+            outputArray->Resize1D(0);
+        return false;
+    }
+    return true;
+}
+
+VIREO_FUNCTION_SIGNATURET(ByteArrayToString, ByteArrayToStringParamBlock)
+{
+    StringRef stringOut = _ParamPointer(StringOut) ? _Param(StringOut) : nullptr;
+    ErrorCluster *errorCluster = _ParamPointer(ErrorInOut);
+    if ((stringOut == nullptr && errorCluster == nullptr) || (errorCluster != nullptr && errorCluster->hasError()))
+        return _NextInstruction();
+    TypedArrayCoreRef byteArray = _Param(ByteArrayIn);
+    TypeRef eltType = byteArray->ElementType();
+    VIREO_ASSERT(eltType->IsA(&TypeCommon::TypeInt8) || eltType->IsA(&TypeCommon::TypeUInt8));
+    IntIndex rank = byteArray->Rank();
+    VIREO_ASSERT(rank == 1);
+    Int32 stringEncoding = _Param(StringEncodingIn);
+    Int32 arrayLength = byteArray->DimensionLengths()[0];
+    InstructionCore* value;
+    if (!IsSupportedEncodingAndUpdateErrorAndMakeOutputArrayEmpty(errorCluster, stringEncoding, stringOut)) {
+        return _NextInstruction();
+    }
+    if (!IsValidUTF8AndUpdateErrorAndMakeOutputArrayEmpty(byteArray->RawBegin(), byteArray->RawBegin() + arrayLength, errorCluster, stringOut)) {
+        return _NextInstruction();
+    }
+    stringOut->Replace1D(0, arrayLength, byteArray->RawBegin(), true);
+    return _NextInstruction();
+}
+
+struct StringToByteArrayParamBlock : InstructionCore
+{
+    _ParamDef(TypedArrayCoreRef, ByteArrayOut);
+    _ParamDef(StringRef, StringIn);
+    _ParamDef(Int32, StringEncodingIn);  // 0 = UTF-8 (only supported)
+    _ParamDef(ErrorCluster, ErrorInOut);
+    NEXT_INSTRUCTION_METHOD()
+};
+
+VIREO_FUNCTION_SIGNATURET(StringToByteArray, StringToByteArrayParamBlock)
+{
+    StringRef stringIn = _Param(StringIn);
+    TypedArrayCoreRef byteArrayOut = _ParamPointer(ByteArrayOut) ? _Param(ByteArrayOut) : null;
+    ErrorCluster *errorCluster = _ParamPointer(ErrorInOut);
+    if (byteArrayOut == nullptr || (errorCluster != nullptr && errorCluster->hasError()))
+        return _NextInstruction();
+    TypeRef eltType = byteArrayOut->ElementType();
+    VIREO_ASSERT(eltType->IsA(&TypeCommon::TypeUInt8));
+    IntIndex rank = byteArrayOut->Rank();
+    VIREO_ASSERT(rank == 1);
+    Int32 stringEncoding = _Param(StringEncodingIn);
+    if (!IsSupportedEncodingAndUpdateErrorAndMakeOutputArrayEmpty(errorCluster, stringEncoding, byteArrayOut)) {
+        return _NextInstruction();
+    }
+    if (!IsValidUTF8AndUpdateErrorAndMakeOutputArrayEmpty(stringIn->RawBegin(), stringIn->RawBegin() + stringIn->Length(), errorCluster, byteArrayOut)) {
```

Comment:
Discussed with @spathiwa.  We are choosing to do validation for StringToByteArray because a LV string might have been gotten by other means and may still contain invalid UTF-8 characters. However, we will populate the ByteArray Output regardless of finding invalid UTF-8 characters. This will allow users to be able to debug also.